artifacts.go

```diff
@@ -58,7 +60,13 @@ func artifactsHandler(indexer Indexer, cacheTime time.Duration) func(w http.Resp
 			return
 		}
 
+		aPackage := packageList[0]
+		if storageIndexer, assert := aPackage.Indexer().(*storage.Indexer); assert {
+			storageIndexer.HijackArtifactsHandler(w, r, aPackage)
+			return
+		}
```

Comment:
Instead of relying on the type of the indexer, maybe we could add some value in the package that helps knowing if the resource is local or remote. Each indexer would be in charge of setting this value accordingly.
With this maybe less refactor is needed, and other future indexers could have redirects too.

```suggestion
		if p := packageList[0]; p.Remote; {
			packages.ServeRedirect(w, r, p)
			return
		}
```
Or it could be a pointer to a resolver that implements the `Hijack*Handler` methods and can be used to resolve the redirect.
```suggestion
		if p := packageList[0]; p.RemoteResolver != nil; {
			packages.ServeRedirect(w, r, p)
			return
		}
```
Perhaps this logic could also be in `packages.ServePackage`, so the handlers remain the same.

Reply:
`RemoteResolver` would be a public field, which I wouldn't expose if it isn't necessary. It might be confusing for other projects using it. Today it's just package-storage-infra, but maybe in the future, we will use it also in elastic-package.

I'm not sure whether depending on Local/Remote boolean property will be flexible enough for future purposes. If we decide to support multiple Remote indexers, we will have to introduce another field.

WDYT?

Reply:
> `RemoteResolver` would be a public field, which I wouldn't expose if it isn't necessary. It might be confusing for other projects using it. Today it's just package-storage-infra, but maybe in the future, we will use it also in elastic-package.

Well, with current implementation the indexer is being exposed with the new `Indexer()` method, what also includes the `Hijack` methods, apart of other things.

My proposal would only expose what is needed in a package to decide if it has to be served locally or as a redirect, and if it is with a redirect, how to do it. This is information that could make sense to have in the Package struct. This would also avoid the need of having indexer-specific logic in handlers.

If preferred, we could make `RemoteResolver` a method instead of a property.